example/build_tools/bazel_rules/jar_scanner/rule.bzl

```diff
@@ -0,0 +1,93 @@
+# jar_scanner_aspect called by py_build_commands.py is intended for scan
+# exactly one JAR file, typically ones exposed by scala_import(...),
+# and generates JSON file listing the class names used by bzl-gen-build.
+
+def _jar_scanner_impl(target, ctx):
+    label = str(target.label)
+    name = target.label.name
+    if ((not target.label.workspace_name.startswith("_main~maven~maven")) and
+        (not label.endswith("proto_java")) and
+        (not label.endswith("proto_scala"))):
+        return []
+
+    # Make sure the rule has a srcs attribute.
+    out = ctx.actions.declare_file("%s_jar_scanner.json" % (target.label.name))
+    files = ctx.rule.files
+    all_jars = []
+
+    # For protobuf-generated targets, we end up with multiple JAR files
+    # returned by info.files, so here I am manually narrowing down to exactly 1 JAR file.
+    if hasattr(files, "deps"):
+        info = target[DefaultInfo]
+
+        # this is a java_proto_library
```

Comment:
this comment isn't true is it? It may be true I guess.

Reply:
Probably not. Removed in d22a6fab63f803b69a622d183b9bfe199ebedd20.